source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,95 @@ Find a Document
 ===============
 
 .. default-domain:: mongodb
+
+You can retrieve a single document from your collection with the
+``*Collection.FindOne()`` method. 
+
+..
+   method
+
+The ``FindOne()`` method takes a query document that specifies the
+subset of your collection that you would like to retrieve a document from. If you
+pass the ``FindOne()`` method an empty document, MongoDB will consider your
+entire collection when retrieving your document. For more information on documents
+in the MongoDB Go driver, see our
+:doc:`guide on data formats </fundamentals/data-formats>`.
+
+You must pass the ``FindOne()`` method a ``context.Context`` interface type
+value. For more information on the ``context.Context`` interface and the
+``context`` package, see our :doc:`guide on context </fundamentals/context>`.
+
+..
+   options
+
+You can specify options that modify the behavior of your ``FindOne()`` method.
+One example of an option is **sort**, which orders the subset of your collection
+that matches your query before the first document is retrieved. Another example
+of an option is a **projection**, which specifies or restricts the fields in
+your returned document. You can specify additional options by passing the
+``FindOne()`` method a pointer to a ``FindOneOptions`` struct. You
+can configure a ``FindOneOptions`` struct by using setter methods such as the
+``*FindOneOptions.SetProjection()`` method.
+
+..
+   return
+
+If the query filter you specified in your ``FindOne()`` method call matches one
+or more documents in the collection, it returns the first matched result. The
+return type ``SingleResult`` represents a single document returned from a query
+to a MongoDB instance. You can call the ``*SingleResult.Decode()`` method
+to unmarshall a ``SingleResult`` value into a variable. Pass the ``Decode()`` 
+method a pointer to the variable that you would like to contain your returned
+document. 
+
+..
+   error 
+
+When you call the ``Decode()`` method to unmarshall your result, the method
+returns an ``error`` interface type which can contain one of the following
+values: 
+
+- ``nil`` if a document matched your query, and there were no errors retrieving
+  the document. 
+- ``ErrNoDocuments`` if no documents matched your query. 
+- If there was an error retrieving your document, the ``Decode()`` method
+  returns the error. 
```

Comment:
In terms of error handling, it may be worth mentioning how errors from the initial operation (e.g. find) will be propagated through methods like `Decode`. In other words, if the find itself fails, we'll return a `SingleResult`, but the `Decode` call will return that error without doing anything else. This allows for fluent API calls like `coll.FindOne(...).Decode()` without having to do an additional error check between the `FindOne` and `Decode` calls.

Reply:
Thanks for the note. I have rewritten this list item to explicitly state that errors from the ``FindOne()`` method call are propagated to ``Decode()`` and ``Decode()`` returns the error. Fix: https://github.com/mongodb/docs-golang/pull/2/commits/0f91e0182facc6e7ed14d46bd00c757cc40768f6